extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -801,11 +815,11 @@ private static HttpServerOptions createSslOptions(HttpBuildTimeConfig buildTimeC
         } else if (keyStoreFile.isPresent()) {
             KeyStoreOptions options = createKeyStoreOptions(
                     keyStoreFile.get(),
-                    keystorePassword,
+                    keyStorePassword.orElse("password"),
```

Comment:
Not sure about injecting "password" as default. I would fail rather than using something obvious.

Reply:
Hi @cescoffier It is because I'd like to avoid breaking some existing code, on `main` it is
```
@ConfigItem(defaultValue = "password")
public String keyStorePassword;
```
But in this PR, for `keyStorePassword` be tried from `CredentialsProvider` as well, I made it
```
@ConfigItem(defaultValueDocumentation = "password")
public Optional<String> keyStorePassword;
```

so that `keyStorePassword.orElse("password")` only preserves the current behavior. 
Yeah it would be worth just removing a default `password` value - but I think we'd better have it done in a dedicated PR

Reply:
Ah I see, yes, I regret having done that choice 3 years ago... 
I think it's something we need to change.

Reply:
@cescoffier Sure, I can follow up with another PR, as technically it would be a breaking change, though in practice not really :-), no one would use such a value in production